### PR TITLE
Bookmark feature - UI

### PR DESCRIPTION
**Pull request by shaunajisc**
The purpose of this PR was to create a feature on the UI to allow users to bookmark a publication, and display a list of the user's bookmarked publications. 

This PR contains amendments to the API to return the relevant information on the bookmarked publications, additions to the publication page to display the interactive bookmark, the creation of a new page for bookmarks, a component for each bookmarked publication, additions to the nav bar, and a few tweaks to other bits on the 'My Publications' page so that the design is consistent. 

## Acceptance Criteria:
- An interactive bookmark is displayed on each live publication page if the user isn't an author or co-author
- The bookmark icon is filled in when the user has previously bookmarked the publication and is just an outline when there is no bookmark under that user and publication
- an authenticated user can access only their own bookmarked publications 
- users can remove bookmarks from the 'My Bookmarks' page 
- users can access their bookmarked publications using a drop down accordion on the account section

### Inactive bookmark:
Screenshot 2022-05-10 at 11 32 58

### Active Bookmark: 
Screenshot 2022-05-10 at 11 33 51

### 'My bookmarks' page:
Screenshot 2022-05-10 at 11 35 22

### Amended navbar: 
Screenshot 2022-05-10 at 11 36 39

### CODE REVIEW

File: ui/src/config/urls.ts [hunk 1]
```diff
@@ -106,6 +106,13 @@ const urls = {
         keywords: [],
         canonical: `${base.host}/account`
     },
+    myBookmarks: {
+        path: '/my-bookmarks',
+        title: `${base.title}`,
+        description: '',
+        keywords: [],
+        canonical: `${base.host}/authors`
```
Comment (AshleyRedmanJisc) [inline ui/src/config/urls.ts:114]: Think this needs updating to `${base.host}/my-bookmarks`

**Review by AshleyRedmanJisc — CHANGES_REQUESTED** [review]
One comment, but pulled local & tested, works well 👍🏼

File: ui/src/pages/my-bookmarks.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    let user: Interfaces.User | null = null;
+
+    let usersBookmarks: Interfaces.BookmarkedPublicationsData[] = [];
+
+    try {
+        const response = await api.get(`${Config.endpoints.bookmarks}`, token);
+        usersBookmarks = response.data;
+    } catch (err) {
+        console.log('err');
```
Comment (tomgiddingsjisc) [inline ui/src/pages/my-bookmarks.tsx:25]: Not sure if the intention is to output the error itself: If so, just needs the quotes removing.

**Review by tomgiddingsjisc — CHANGES_REQUESTED** [review]
One small comment to add, but looks good otherwise!

File: ui/src/pages/my-bookmarks.tsx (hunk 2 above)
Comment (shaunajisc) [reply ui/src/pages/my-bookmarks.tsx:25]: Ah yes - my bad!

File: ui/src/config/urls.ts (hunk 1 above)
Comment (shaunajisc) [reply ui/src/config/urls.ts:114]: good spot - I'll amend

**Review by tomgiddingsjisc — APPROVED** [review]
lgtm!

**Review by AshleyRedmanJisc — APPROVED** [review]
lgtm

**Pull request merged**